evap/static/ts/src/staff-questionnaire-form.ts

```diff
@@ -0,0 +1,147 @@
+        const questionType = saneParseInt(target.value);
+        const questionTypeCell = target.closest("td.question-type");
+        if (!questionTypeCell) return;
+
+        const checkboxes = questionTypeCell.querySelectorAll("input[type=checkbox]");
+
+        if (questionType === QUESTION_TYPE_TEXT || questionType === QUESTION_TYPE_HEADING) {
+            this.disableAllCheckboxes(checkboxes);
+            return;
+        }
+        // Check if this is a dropout questionnaire before enabling checkboxes
+        const questionnaireType = saneParseInt(this.questionnaireTypeSelect.value);
+
+        if (questionnaireType === QUESTIONNAIRE_TYPE_DROPOUT) {
+            checkboxes.forEach(checkbox => {
+                const checkboxElement = checkbox as HTMLInputElement;
+                if (checkboxElement.classList.contains("counts-for-grade-checkbox")) {
+                    this.disableAndUncheckCheckbox(checkboxElement);
+                } else {
+                    this.enableAndCheckCheckbox(checkboxElement);
+                }
+            });
+        } else {
+            this.enableAllCheckboxes(checkboxes);
+        }
+    };
+
+    private handleQuestionnaireTypeChange = (): void => {
+        const selectedType = saneParseInt(this.questionnaireTypeSelect.value);
+        const countsForGradeCheckboxes = document.querySelectorAll(".counts-for-grade-checkbox");
+
+        if (selectedType === QUESTIONNAIRE_TYPE_DROPOUT) {
+            countsForGradeCheckboxes.forEach(checkbox => {
+                const checkboxElement = checkbox as HTMLInputElement;
+                const questionTypeCell = checkboxElement.closest("td.question-type");
+                assertDefined(questionTypeCell);
+
+                const questionTypeSelect = selectOrError<HTMLSelectElement>("select", questionTypeCell);
+
+                if (questionTypeSelect.value === "") {
+                    return;
+                }
+
+                this.disableAndUncheckCheckbox(checkboxElement);
+            });
+        } else {
+            countsForGradeCheckboxes.forEach(checkbox => {
+                const checkboxElement = checkbox as HTMLInputElement;
+                const questionTypeCell = checkboxElement.closest("td.question-type");
+                assertDefined(questionTypeCell);
+
+                const questionTypeSelect = selectOrError<HTMLSelectElement>("select", questionTypeCell);
+
+                if (questionTypeSelect.value === "") {
+                    return;
+                }
+
+                const questionType = saneParseInt(questionTypeSelect.value);
+                if (questionType === QUESTION_TYPE_TEXT || questionType === QUESTION_TYPE_HEADING) {
+                    this.disableAndUncheckCheckbox(checkboxElement);
+                } else {
+                    this.enableAndCheckCheckbox(checkboxElement);
+                }
+            });
+        }
+    };
+
+    public initialize = (): void => {
```

Comment:
Seems to me that initialization is the purpose of the constructor, having a method `initialize` is smelly. The single user of this class now does
```js
            const staffQuestionnaireForm = new StaffQuestionnaireForm("question_table");
            staffQuestionnaireForm.initialize();
```
Is there an advantage to this, or should this just be private and called from the constructor, or possibly even be inlined into the constructor?